storage/netapp/netapp_e_storagepool.py

```diff
@@ -0,0 +1,888 @@
+                                          exact_drive_count=exact_drive_count, raid_level=raid_level):
+            return False
+        # TODO: this assumes candidate_set is all the same size- if we want to allow wastage, need to update to use min size of set
+        if min_capacity_bytes != None and self._calculate_usable_capacity(int(candidate_set[0]['rawCapacity']),
+                                                                          len(candidate_set),
+                                                                          raid_level=raid_level) < min_capacity_bytes:
+            return False
+
+        return True
+
+    def _calculate_usable_capacity(self, disk_size_bytes, disk_count, raid_level=None):
+        if raid_level in [None, 'raid0']:
+            return disk_size_bytes * disk_count
+        if raid_level == 'raid1':
+            return (disk_size_bytes * disk_count) / 2
+        if raid_level in ['raid3', 'raid5']:
+            return (disk_size_bytes * disk_count) - disk_size_bytes
+        if raid_level in ['raid6', 'raidDiskPool']:
+            return (disk_size_bytes * disk_count) - (disk_size_bytes * 2)
+        raise Exception("unsupported raid_level: %s" % raid_level)
+
+    def _is_drive_count_valid(self, drive_count, min_drive_count=0, exact_drive_count=None, raid_level=None):
+        if exact_drive_count and exact_drive_count != drive_count:
+            return False
+        if raid_level == 'raidDiskPool':
+            if drive_count < 11:
+                return False
+        if raid_level == 'raid1':
+            if drive_count % 2 != 0:
+                return False
+        if raid_level in ['raid3', 'raid5']:
+            if drive_count < 3:
+                return False
+        if raid_level == 'raid6':
+            if drive_count < 4:
+                return False
+        if min_drive_count and drive_count < min_drive_count:
+            return False
+
+        return True
+
+    def get_storage_pool(self, storage_pool_name):
+        # global ifilter
+        self.debug("fetching storage pools")
+        # map the storage pool name to its id
+        try:
+            (rc, resp) = request(self.api_url + "/storage-systems/%s/storage-pools" % (self.ssid),
+                                 headers=dict(Accept="application/json"), url_username=self.api_usr,
+                                 url_password=self.api_pwd, validate_certs=self.validate_certs)
+        except Exception:
+            err = get_exception()
+            rc = err.args[0]
+            if rc == 404 and self.state == 'absent':
+                self.module.exit_json(
+                    msg="Storage pool [%s] did not exist." % (self.name))
+            else:
+                err = get_exception()
+                self.module.exit_json(
+                    msg="Failed to get storage pools. Array id [%s].  Error[%s]. State[%s]. RC[{3}]." % (
```

Comment:
MUSTFIX: E:454,24: Too many arguments for format string (too-many-format-args)

(mixing %s and {} format strings here, intentional?)